pymsis/utils.py

```diff
@@ -92,19 +150,34 @@ def _load_f107_ap_data() -> dict[str, npt.NDArray]:
     # so we can't just go back in line lengths)
     with _F107_AP_PATH.open() as fin:
         with BytesIO() as fout:
+            last_obs_date: np.datetime64 | None = None
             for line in fin:
-                if "PRM" in line:
-                    # We don't want the monthly predicted values
-                    continue
-                if ",,,,,,,," in line:
-                    # We don't want lines with missing values
+                if "PRM" in line or ",,,,,,,," in line:
+                    # We don't want the monthly predicted values or missing values
                     continue
+                if ",OBS," in line:
+                    # Capture last observed date
+                    last_obs_date = np.datetime64(line.split(",")[0])
                 fout.write(line.encode("utf-8"))
             fout.seek(0)
             arr = np.loadtxt(
                 fout, delimiter=",", dtype=dtype, usecols=usecols, skiprows=1
             )  # type: ignore
 
+    # Check if the file needs to be refreshed after parsing
+    if last_obs_date is not None:
+        file_mod_time = datetime.fromtimestamp(os.path.getmtime(_F107_AP_PATH))
```

Comment:
Do we need file modification time? I think we should just leave that out and only rely on the requested date being greater than the observations we have.